examples/Basic/gui_table.py

```diff
@@ -0,0 +1,97 @@
+"""
+This example demonstrates how to make a graphical interface, and uses
+a random number generator to simulate data so that it does not require
+an instrument to use. In particular, this example shows how to display
+data in tabular format.
+
+Run the program by changing to the directory containing this file and calling:
+
+python gui_table.py
+
+"""
+
+import sys
+import random
+from time import sleep
+from pymeasure.experiment import Procedure, IntegerParameter, Parameter, FloatParameter
+from pymeasure.experiment import Results, unique_filename
+from pymeasure.display.Qt import QtWidgets
+from pymeasure.display.windows import ManagedWindowBase
+from pymeasure.display.widgets import TableWidget, LogWidget
+
+import logging
+log = logging.getLogger('')
+log.addHandler(logging.NullHandler())
+
+
+class TestProcedure(Procedure):
+
+    iterations = IntegerParameter('Loop Iterations', default=10)
+    delay = FloatParameter('Delay Time', units='s', default=0.2)
+    seed = Parameter('Random Seed', default='12345')
+
+    DATA_COLUMNS = ['Iteration', 'Random Number']
+
+    def startup(self):
+        log.info("Setting up random number generator")
+        random.seed(self.seed)
+
+    def execute(self):
+        log.info("Starting to generate numbers")
+        for i in range(self.iterations):
+            data = {
+                'Iteration': i,
+                'Random Number': random.random()
+            }
+            log.debug("Produced numbers: %s" % data)
+            self.emit('results', data)
+            self.emit('progress', 100 * i / self.iterations)
+            sleep(self.delay)
+            if self.should_stop():
+                log.warning("Catch stop command in procedure")
+                break
+
+    def shutdown(self):
+        log.info("Finished")
+
+
+class ManagedWindowWithTable(ManagedWindowBase):
+    def __init__(self, procedure_class, **kwargs):
+        self.log_widget = LogWidget("Experiment Log")
+        self.table_widget = TableWidget("Experiment Table",
+                                        procedure_class.DATA_COLUMNS,
+                                        by_column=True,
+                                        )
+        if "widget_list" not in kwargs:
+            kwargs["widget_list"] = ()
+        kwargs["widget_list"] = kwargs["widget_list"] + (self.table_widget, self.log_widget)
+
+        super().__init__(procedure_class, **kwargs)
```

Comment:
Why this intermediate class? I'd be inclined to think you can do everything in the MainWindow?

Reply:
Thanks, you are right. I will remove it.

Reply:
Done in commit https://github.com/pymeasure/pymeasure/pull/771/commits/8081830a2de49d1b01b78fe9b35435252bbc644f